Extensions/ArtifactEngine/Store/artifactItemStore.ts

```diff
@@ -47,13 +62,22 @@ export class ArtifactItemStore {
         return undefined;
     }
 
-    public updateState(item: models.ArtifactItem, state: models.TicketState) {
+    public updateState(item: models.ArtifactItem, state: models.TicketState, location?: boolean) {
         var processedItem = this._downloadTickets.find(x => x.artifactItem.path === item.path);
         if (processedItem) {
             processedItem.state = state;
+            if (location) {
+                processedItem.downloadedFromCache = location;
+            }
             if (state != models.TicketState.InQueue && state != models.TicketState.Processing) {
                 processedItem.finishTime = new Date();
             }
+            if ((item.downloadedFileHash && item.downloadedFileHash === item.fileHash) || !item.fileHash) {
+                return true;
+            }
+            else {
+                return false;
```

Comment:
it will not return false in cases where no cache is involved. It is the validation step that checks whether the downloaded content is same as the one which we wanted to download, so it returns true as long as the contents are same.